src/Agda2Hs/Compile/Imports.hs

```diff
@@ -20,7 +20,7 @@ import Agda2Hs.Compile.Types
 import Agda2Hs.Compile.Utils
 import Agda2Hs.HsUtils
 
-type ImportSpecMap = Map (Hs.Name ()) (Set (Hs.Name ()))
+type ImportSpecMap = Map (Hs.Name (), Hs.Namespace ()) (Set (Hs.Name (), Hs.Namespace ()))
```

Comment:
Perhaps you could introduce a new type that packages a `Name` and a `Namespace`? These tuples are a bit too easy to get confused about.

Reply:
This is done now:)